File: C/122. 买卖股票的最佳时机 II.c
int maxProfit(int* prices, int pricesSize){
    int l =0,r=0,count =0;
    while(r<pricesSize){
        if(l==r){
            r++;
        }else if(prices[r]<prices[l]){
            l=r;
        }else {
            count+=(prices[r++]-prices[l++]);
        }
    }
    return count;
}

File: doublePoint/122. 买卖股票的最佳时机 II.js
/**
 * @param {number[]} prices
 * @return {number}
 */
 var maxProfit = function(prices) {
  let l=0,r=1,count=0;
  while(r<prices.length){
      if(prices[l]>prices[r]){
          l=r;
      }else if(l===r){
          r++;
      } else {
          count+=(prices[r++]-prices[l++]);
      }
  }
  return count;
};